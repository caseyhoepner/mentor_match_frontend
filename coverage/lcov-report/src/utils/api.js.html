<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/utils/api.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/utils</a> api.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>22/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">73.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/30</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { cleanData } from './helper';
&nbsp;
export const fetchMentors = async () =&gt; {
  const url = 'https://mentor-match-api.herokuapp.com/api/v1/mentors';
  const response = await fetch(url);
  const mentors = await response.json();
  return cleanData(mentors);
};
&nbsp;
export const postMentor = async (mentor) =&gt; {
  const url = 'https://mentor-match-api.herokuapp.com/api/v1/mentors';
  const response = await fetch(url, {
    method: "POST",
    headers: { "Content-Type": "application/json" }, 
    accept: "application/json",
    body: JSON.stringify(mentor)
  });
  const status = await response.json();
  console.log(status);
};
&nbsp;
export const patchMentor = async (mentor) =&gt; {
   const url = `https://mentor-match-api.herokuapp.com/api/v1/mentors/${mentor.id}`;
   const response = await fetch(url, {
    method: "PATCH",
     headers: { "Content-Type": "application/json", "Accept": "application/json" },
     body: JSON.stringify(mentor)
   });
   const status = await response.json();
   console.log(status);
 };
&nbsp;
 export const fetchStudents = async () =&gt; {
  const url = 'https://mentor-match-api.herokuapp.com/api/v1/students';
  const response = await fetch(url);
  const students = <span class="cstat-no" title="statement not covered" >await response.json();</span>
<span class="cstat-no" title="statement not covered" >  return cleanData(students);</span>
};
&nbsp;
export const postRelationship = <span class="fstat-no" title="function not covered" >as</span>ync (studentId, mentorId) =&gt; {
  const url = <span class="cstat-no" title="statement not covered" >'https://mentor-match-api.herokuapp.com/api/v1/student_mentors';</span>
  const response = <span class="cstat-no" title="statement not covered" >await fetch(url, {</span>
    method: "POST",
    headers: { "Content-Type": "application/json" }, 
    accept: "application/json",
    body: JSON.stringify({
      student_id: studentId,
      mentor_id: mentorId
    })
  });
  const status = <span class="cstat-no" title="statement not covered" >await response.json();</span>
<span class="cstat-no" title="statement not covered" >  console.log(status);</span>
};
&nbsp;
export const fetchRelationships = async () =&gt; {
  const url = 'https://mentor-match-api.herokuapp.com/api/v1/student_mentors';
  const response = await fetch(url);
  const relationships = <span class="cstat-no" title="statement not covered" >await response.json();</span>
<span class="cstat-no" title="statement not covered" >  return cleanData(relationships);</span>
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jan 03 2019 17:48:11 GMT-0700 (MST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
