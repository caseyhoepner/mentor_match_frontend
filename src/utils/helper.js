export const cleanMentors = (mentors) => {
  return mentors.data.map(mentor => {
    return mentor.attributes;
  })
}