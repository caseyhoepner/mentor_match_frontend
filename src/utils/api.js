import { cleanMentors } from './helper';

export const fetchMentors = async () => {
  const url = 'https://mentor-match-api.herokuapp.com/api/v1/mentors';
  const response = await fetch(url);
  const mentors = await response.json();
  return cleanMentors(mentors);
};

export const postMentor = async (mentor) => {
  const url = 'https://mentor-match-api.herokuapp.com/api/v1/mentors';
  const response = await fetch(url, {
    method: "POST",
    headers: { "Content-Type": "application/json" }, 
    accept: "application/json",
    body: JSON.stringify(mentor)
  });
  const status = await response.json();
  console.log(status);
};

export const patchMentor = async (mentor) => {
   const url = `https://mentor-match-api.herokuapp.com/api/v1/mentors/${mentor.id}`;
   const response = await fetch(url, {
    method: "PATCH",
     headers: { "Content-Type": "application/json", "Accept": "application/json" },
     body: JSON.stringify(mentor)
   });
   const status = await response.json();
   console.log(status);
 };
