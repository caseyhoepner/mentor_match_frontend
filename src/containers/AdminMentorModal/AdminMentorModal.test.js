import React from 'react';
import { shallow } from 'enzyme';
import { AdminMentorModal, mapStateToProps, mapDispatchToProps } from './';
import * as API from '../../utils/api';

describe('AdminMentorModal', () => {
  let wrapper;
  let mockFunction = jest.fn()
  let mockFunc2 = jest.fn()
  let mockFunc3 = jest.fn()
  let mockFunc4 = jest.fn()
  let mockFunc5 = jest.fn()
  let mockFunc6 = jest.fn()
  let mockHistory = []
  let mockMentor = {
    id: 6,
    name: 'Menty the Mentor',
    email: 'menty@mentor.com',
    city: 'Denver',
    state: 'CO',
    country: 'USA',
    slack_username: '@Menty',
    matched: true,
    active: true,
    pronouns: 'she/her/hers',
    current_title: 'Head Mentor',
    current_employer: 'Department of Mentors',
    background: 'I am a mentor!',
    industries: ['Civic Tech'],
    ways_to_mentor: ['Coffee Meetings'],
    expertise_tech: ['React.js'],
    expertise_non_tech: ['Parenting'],
    mentee_capacity: '2',
    identity_preference: ['female-identifying'],
    meeting_location: 'Turing'
  }
  let mockStudents = [
    { name: 'Casey', id: 1, active: true, matched: false },
    { name: 'Alex', id: 2, active: false, matched: true }
  ]
  
  let mockRelationships = [
    {
      attributes: {
        mentor_id: 1,
        student_id: 5,
        active: true
      }
    }
  ]

  beforeEach(() => {
    wrapper = shallow(<AdminMentorModal 
      currentMentor={mockMentor} 
      students={mockStudents} 
      modalInfo={mockMentor}
      relationships={mockRelationships}
      isEditable={false}
      addModalMentees={mockFunction}
      updateChangedMentor={mockFunc2}
      setMentorModal={mockFunc3}
      openEditMentor={mockFunc4}
      makeStudentInactive={mockFunc5}
      retrieveRelationships={mockFunc6}
    />)
  });

  it('should match the snapshot with props and when it is not editable', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot with props and when it is editable', () => {
    wrapper = shallow(<AdminMentorModal 
      currentMentor={mockMentor} 
      students={mockStudents} 
      modalInfo={mockMentor}
      relationships={mockRelationships}
      isEditable={true}
      addModalMentees={mockFunction}
    />)
    expect(wrapper).toMatchSnapshot()
  });

  describe('handleClick function', () => {
    it('should update state to be editable and add the mentor from props in state', () => {
      wrapper = shallow(<AdminMentorModal 
        modalInfo={mockMentor} 
        openEditMentor={mockFunction} 
        history={mockHistory}  
        students={mockStudents} 
        relationships={mockRelationships}
        addModalMentees={mockFunction}
      />)
      let mockEvent = { target: { name: 'Edit' } }

      wrapper.instance().handleClick(mockEvent)
      expect(mockFunction).toHaveBeenCalledWith(true)
      expect(wrapper.state().currentMentor).toEqual(mockMentor)
    });

    it('should fire setMentorModal function from props', () => {
      let mockFunc1 = jest.fn()
      let mockFunc2 = jest.fn()
      let mockEvent = { target: { name: 'Submit Changes' } }
      wrapper = shallow(<AdminMentorModal 
        modalInfo={mockMentor}
        setMentorModal={mockFunc1} 
        updateChangedMentor={mockFunc2} 
        openEditMentor={mockFunction} 
        history={mockHistory} 
        students={mockStudents}
        relationships={mockRelationships}
        addModalMentees={mockFunction}
      />)
      
      wrapper.instance().handleClick(mockEvent)
      expect(mockFunc1).toHaveBeenCalled()
    })

    it('should fire updateChangedMentor function from props with proper params', () => {
      let mockFunc1 = jest.fn()
      let mockFunc2 = jest.fn()
      let mockEvent1 = { target: { name: 'Edit' } }
      let mockEvent2 = { target: { name: 'Submit Changes' } }
      wrapper = shallow(<AdminMentorModal 
        modalInfo={mockMentor}
        setMentorModal={mockFunc1} 
        updateChangedMentor={mockFunc2}
        openEditMentor={mockFunction} 
        history={mockHistory} 
        students={mockStudents}
        relationships={mockRelationships}
        addModalMentees={mockFunction}
      />)
      
      wrapper.instance().handleClick(mockEvent1)
      wrapper.instance().handleClick(mockEvent2)
      expect(mockFunc2).toHaveBeenCalledWith(mockMentor)
    });

    it('should update state to turn editability off and zero out the currentMentor in state', () => {
      let mockFunc1 = jest.fn()
      let mockFunc2 = jest.fn()
      let mockEvent1 = { target: { name: 'Edit' } }
      let mockEvent2 = { target: { name: 'Submit Changes' } }
      wrapper = shallow(<AdminMentorModal 
        modalInfo={mockMentor}
        setMentorModal={mockFunc1} 
        updateChangedMentor={mockFunc2} 
        openEditMentor={mockFunction}
        history={mockHistory} 
        students={mockStudents}
        relationships={mockRelationships}
        addModalMentees={mockFunction}
      />)
        
      wrapper.instance().handleClick(mockEvent1)
      wrapper.instance().handleClick(mockEvent2)
      expect(mockFunction).toHaveBeenCalledWith(false)
      expect(wrapper.state().currentMentor).toEqual({})
    });

    it('should reset redux and state when the Exit/X button is clicked', () => {
      let mockFunc1 = jest.fn()
      let mockEvent1 = { target: { name: 'Edit' } }
      let mockEvent2 = { target: { name: 'Exit' } }
      wrapper = shallow(<AdminMentorModal 
        modalInfo={mockMentor}
        setMentorModal={mockFunc1}
        openEditMentor={mockFunction}
        history={mockHistory} 
        students={mockStudents}
        relationships={mockRelationships}
        addModalMentees={mockFunction}
      />)

      wrapper.instance().handleClick(mockEvent1)
      expect(wrapper.state()).toEqual({ currentMentor: mockMentor, menteeToAssign: '', mentees: [] })
      wrapper.instance().handleClick(mockEvent2)
      expect(mockFunc1).toHaveBeenCalled()
      expect(mockFunction).toHaveBeenCalledWith(false)
      expect(wrapper.state()).toEqual({ currentMentor: {}, menteeToAssign: '', mentees: [] })
    });
  });

  describe('handleChange function', () => {
    it('should update menteeToAssign in state when a value is changed', () => {
      let mockOriginalState = {
        menteeToAssign: 'Dan'
      }

      let mockEvent = { target: { name: 'menteeToAssign', value: 'Halle Berry' } }

      wrapper.instance().setState(mockOriginalState)
      wrapper.instance().handleChange(mockEvent)
      expect(wrapper.state().menteeToAssign).toEqual('Halle Berry')
    });
  });

  describe('getStudent function', () => {
    it('should return the student that matches the menteeToAssign value in state', () => {
      let mockStudents = [{ name: 'Rosa Diaz' },{ name: 'Ray Holt' }]
      wrapper.instance().setState({ menteeToAssign: 'Ray Holt' })
      let result = wrapper.instance().getStudent(mockStudents)
      expect(result).toEqual({ name: 'Ray Holt' })
    });
  });

  describe('assignMentee function', () => {
    it('should fire the getStudent function with the correct params', async () => {
      let spy = jest.spyOn(wrapper.instance(), 'getStudent')
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(spy).toHaveBeenCalledWith(mockStudents)
    });

    it('should fire patchMentor fetch call with correct params', async () => {
      let spy = jest.spyOn(API, 'patchMentor')
      let expected = {
        id: 6,
        name: 'Menty the Mentor',
        email: 'menty@mentor.com',
        city: 'Denver',
        state: 'CO',
        country: 'USA',
        slack_username: '@Menty',
        matched: true,
        active: true,
        pronouns: 'she/her/hers',
        current_title: 'Head Mentor',
        current_employer: 'Department of Mentors',
        background: 'I am a mentor!',
        industries: ['Civic Tech'],
        ways_to_mentor: ['Coffee Meetings'],
        expertise_tech: ['React.js'],
        expertise_non_tech: ['Parenting'],
        mentee_capacity: 1,
        identity_preference: ['female-identifying'],
        meeting_location: 'Turing'
      }
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(spy).toHaveBeenCalledWith(expected)
    });

    it('should fire updateChangedMentor with correct params', async () => {
      let expected = {
        id: 6,
        name: 'Menty the Mentor',
        email: 'menty@mentor.com',
        city: 'Denver',
        state: 'CO',
        country: 'USA',
        slack_username: '@Menty',
        matched: true,
        active: true,
        pronouns: 'she/her/hers',
        current_title: 'Head Mentor',
        current_employer: 'Department of Mentors',
        background: 'I am a mentor!',
        industries: ['Civic Tech'],
        ways_to_mentor: ['Coffee Meetings'],
        expertise_tech: ['React.js'],
        expertise_non_tech: ['Parenting'],
        mentee_capacity: 1,
        identity_preference: ['female-identifying'],
        meeting_location: 'Turing'
      }
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(mockFunc2).toHaveBeenCalledWith(expected)
    });

    it('should fire setMentorModal with correct params', async () => {
      let expected = {
        id: 6,
        name: 'Menty the Mentor',
        email: 'menty@mentor.com',
        city: 'Denver',
        state: 'CO',
        country: 'USA',
        slack_username: '@Menty',
        matched: true,
        active: true,
        pronouns: 'she/her/hers',
        current_title: 'Head Mentor',
        current_employer: 'Department of Mentors',
        background: 'I am a mentor!',
        industries: ['Civic Tech'],
        ways_to_mentor: ['Coffee Meetings'],
        expertise_tech: ['React.js'],
        expertise_non_tech: ['Parenting'],
        mentee_capacity: 1,
        identity_preference: ['female-identifying'],
        meeting_location: 'Turing'        
      }
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(mockFunc3).toHaveBeenCalledWith(expected)
    });

    it('should fire postRelationship fetch call with correct params', async () => {
      let spy = jest.spyOn(API, 'postRelationship')
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(spy).toHaveBeenCalledWith(1, 6)
    });

    it('should fire makeStudentInactive function with correct params', async () => {
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(mockFunc5).toHaveBeenCalledWith(1)
    });

    it('should fire patchStudent fetch call with correct params', async () => {
      let spy = jest.spyOn(API, 'patchStudent')
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(spy).toHaveBeenCalledWith({ name: 'Casey', id: 1, active: true, matched: false })
    });

    it('should fire retrieveRelationships function', async () => {
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(mockFunc6).toHaveBeenCalled()
    });

    it('should fire getMentees function', async () => {
      let spy = jest.spyOn(wrapper.instance(), 'getMentees')
      wrapper.instance().setState({ menteeToAssign: 'Casey' })
      await wrapper.instance().assignMentee()
      expect(spy).toHaveBeenCalled()
    });
  });

  describe('getMenteeIcons function', () => {
    it('should return the correct amount of capacity icons', () => {
      let result = wrapper.instance().getMenteeIcons(5)
      expect(result.length).toEqual(5)
    });
  });

  describe('getList function', () => {
    it('should return an array the same length as the one passed in', () => {
      let result = wrapper.instance().getList(['React.js','Ember.js'])
      expect(result.length).toEqual(2)
    });
  });

  describe('getPreferencesIcons function', () => {
    it('should return an array the same length as the one passed in', () => {
      let result = wrapper.instance().getPreferencesIcons(['Veteran', 'lgbtq+'])
      expect(result.length).toEqual(2)
    });
  });

  describe('getStudentOptions function', () => {
    it('should return array of students', () => {
      let result = wrapper.instance().getStudentOptions()
      expect(result.length).toEqual(2)
    });
  });
});
    
describe('mapStateToProps function', () => {
  it('should return an object with a mentor object, isEditable boolean and a mentors array', () => {
    const mockState = {
      modalInfo: { name: 'Leslie Knope', identity_preference: 'female-identifying' },
      mentors: [ { name: 'Ron Swanson', identity_preference: 'male-identifying' } ],
      isEditable: false,
      relationships: [ { mentor_id: 1, student_id: 2, active: true } ],
      somethingElse: 'W00t!'
    }
    const expected = {
      modalInfo: { name: 'Leslie Knope', identity_preference: 'female-identifying' },
      mentors: [ { name: 'Ron Swanson', identity_preference: 'male-identifying' } ],
      isEditable: false,
      relationships: [ { mentor_id: 1, student_id: 2, active: true } ]
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
 });
});
  
describe('mapDispatchToProps function', () => {
  const mockDispatch = jest.fn()
  let mappedProps;
  
  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });
  
  it('should call dispatch when setMentorModal is called', () =>{
    mappedProps.setMentorModal()
    expect(mockDispatch).toHaveBeenCalled()
  });
  
  it('should call dispatch when updateChangedMentor is called', () => {
    mappedProps.updateChangedMentor()
    expect(mockDispatch).toHaveBeenCalled()
  });
  
  it('should call dispatch when openEditMentor is called', () => {
    mappedProps.openEditMentor(true)
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when makeStudentInactive is called', () => {
    mappedProps.makeStudentInactive()
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when addModalMentees is called', () => {
    mappedProps.addModalMentees()
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when retrieveRelationships is called', () => {
    mappedProps.retrieveRelationships()
    expect(mockDispatch).toHaveBeenCalled()
  });
});