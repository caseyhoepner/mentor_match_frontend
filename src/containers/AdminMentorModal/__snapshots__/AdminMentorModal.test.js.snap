// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMentorModal should match the snapshot with props and when it is editable 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminMentorModal
    currentMentor={
      Object {
        "active": true,
        "background": "I am a mentor!",
        "city": "Denver",
        "country": "USA",
        "current_employer": "Department of Mentors",
        "current_title": "Head Mentor",
        "email": "menty@mentor.com",
        "matched": true,
        "name": "Menty the Mentor",
        "pronouns": "she/her/hers",
        "slack_username": "@Menty",
        "state": "CO",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "amm-modal-hide",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "amm-modal-hide",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AdminMentorModal should match the snapshot with props and when it is not editable 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminMentorModal
    currentMentor={
      Object {
        "active": true,
        "background": "I am a mentor!",
        "city": "Denver",
        "country": "USA",
        "current_employer": "Department of Mentors",
        "current_title": "Head Mentor",
        "email": "menty@mentor.com",
        "matched": true,
        "name": "Menty the Mentor",
        "pronouns": "she/her/hers",
        "slack_username": "@Menty",
        "state": "CO",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "amm-modal-hide",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "amm-modal-hide",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AdminMentorModal should match the snapshot without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminMentorModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "amm-modal-hide",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "amm-modal-hide",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
